aravis: 0.6.4 -> 0.8.21, add myself as maintainer

tpwrules opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

This upgrades Aravis to a supported version. I've tested both the viewer (though it has problems rendering on some desktop environments) and the GStreamer plugin with a handful of USB3 Vision cameras. I have not been able to test with GigE Vision cameras.

I've also added myself as a maintainer as there were none and I have both the need of this program and (some of) the necessary hardware to properly test it.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [X] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

tpwrules committed:
aravis: 0.6.4 -> 0.8.21, add myself as maintainer

Mindavi commented on pkgs/development/libraries/aravis/default.nix:
```diff
@@ -1,85 +1,77 @@
+, enableGstPlugin ? true
+, enableViewer ? true
+, gst_all_1 ? null
+, hicolor-icon-theme ? null
```
Don't add nulls here.

tpwrules replied:
Should the asserts be removed too?

lovesegfault replied:
You can leave the asserts.

Mindavi commented on pkgs/development/libraries/aravis/default.nix:
```diff
@@ -1,85 +1,77 @@
+    longDescription = ''
+      Implements the gigabit ethernet and USB3 protocols used by industrial cameras.
+    '';
+    homepage = "https://aravisproject.github.io/docs/aravis-0.8";
```
I think it makes more sense to just refer to the root url rather than to the docs for the `homepage`.

tpwrules replied:
Turns out the root URL refers to a mostly blank page with no useful links or information. This is the best link that's not the Github repository page.

tpwrules commented:
Thanks for the reviews. I also added the proper `gst-plugins-good` input to fix the viewer not being rendered properly, and removed the icon input as it didn't do anything.
